### ISSUE TITLE
Segmentation Fault when running xupric

### ISSUE THREAD

**Issue by DaVikingMan** [chm46e/sulfer#1]
Issue:
Xupric doesn't run and returns a segmentation fault error. I've installed all of the dependencies that are mentioned on the README

**Comment by veisveer** [comment]
Hello!

Can you supply some more information about the output and your system?
What steps did you take to run the browser?
Could you try running it through valgrind?

I've only tested it in my machine, so changes in architecture and specs may throw it off..

**Comment by DaVikingMan** [comment]
I'm running it on archlinux with an x64 cpu. I had followed the installation instructions to install the program. The error message printed is the following `[1]    3921 segmentation fault (core dumped)  xupric`

EDIT : It runs normally under valgrind but doesn't when ran normally(by ./xupric in build/ or xupric from the terminal)
Memory leak summary by valgrind:
`==4409== LEAK SUMMARY:
==4409==    definitely lost: 17,723 bytes in 55 blocks
==4409==    indirectly lost: 88,588 bytes in 3,644 blocks
==4409==      possibly lost: 210,781 bytes in 1,541 blocks
==4409==    still reachable: 5,079,177 bytes in 56,363 blocks
==4409==                       of which reachable via heuristic:
==4409==                         multipleinheritance: 4,008 bytes in 16 blocks
==4409==         suppressed: 889 bytes in 1 blocks
==4409== Reachable blocks (those to which a pointer was found) are not shown.
==4409== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==4409==
==4409== For lists of detected and suppressed errors, rerun with: -s
==4409== ERROR SUMMARY: 263 errors from 217 contexts (suppressed: 0 from 0)` 

The leaks are probably caused as some parts of the program which are declared on the heap aren't freed afterwards. The segmentation fault may also be due to one of the variables which has already been freed but the program's trying to access it.

**Comment by veisveer** [comment]
Okay, so I tried the build in a garudalinux live iso and was able to reproduce the sigsegv.
So it took some time, but I was able to trace it to the webkit_settings_new_with_settings() library call in frame/frame.c
Found a mistype aka config[conf_charset].i was supposed to be a .s
I have absolutely no clue why it worked in my linuxfromscratch system. I guess it doesn't care:D
I realized the readme is wrong, missing deps and the window sizes in a de. are very "reasonable".

I will roll-out some fixes today. Let me know if it works.
